src/Sylius/Bundle/ChannelBundle/DependencyInjection/Compiler/CompositeChannelContextPass.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\ChannelBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * @author Kamil Kokot <kamil.kokot@lakion.com>
+ */
+final class CompositeChannelContextPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if (!$container->hasDefinition('sylius.context.channel')) {
```

Comment:
TODO: Define that service at `sylius.context.channel.composite` instead and proceed only if `sylius.context.channel` does not exist. If someone would like to provide different implementation, we don't want to add method calls on another service. At the end of the function the alias `sylius.context.channel` -> `sylius.context.channel.composite` should be set